djangocms_versioning/plugin_rendering.py

```diff
@@ -24,14 +27,33 @@ def prefetch_versioned_related_objects(instance, toolbar):
             qs = versionable.content_model.objects.all()
         related_field = getattr(instance, field.name)
         if related_field:
-            filters = {versionable.grouper_field_name: related_field}
+            is_related_manager = isinstance(related_field, models.Manager)
+
+            if is_related_manager:
+                filter_key = '{}__in'.format(versionable.grouper_field_name)
+                filter_values = related_field.all()
+                filters = {filter_key: filter_values}
+            else:
+                filters = {versionable.grouper_field_name: related_field}
             # TODO Figure out grouping values-awareness
             # for extra fields other than hardcoded 'language'
             if "language" in versionable.extra_grouping_fields:
                 filters["language"] = toolbar.request_language
             qs = qs.filter(**filters)
             prefetch_cache = {versionable.grouper_field.remote_field.name: qs}
-            related_field._prefetched_objects_cache = prefetch_cache
+
+            # TODO refine it after understand prefetch in many2many field.
+            # because if `related_field` is ManyRelatedManager, it is temporary,
+            # we can't store `_prefetched_objectes_cache` to it, so I decide to store the
+            # prefetched value to model instance.
+            if is_related_manager:
+                try:
+                    instance._prefetched_objects_cache[field.name] = prefetch_cache
```

Comment:
If you use this instead I believe you'd eliminate the try ... except block entirely. 
```
getattr(instance, '_prefetch_objects_cache', {})[field.name] = prefetch_cache
```

Reply:
Thanks mate